manticore/platforms/linux.py

```diff
@@ -51,15 +51,19 @@ def tell(self, *args):
         return self.file.tell(*args)
     def seek(self, *args):
         return self.file.seek(*args)
-    def write(self, *args):
-        return self.file.write(*args)
+    def write(self, buf):
+        for c in buf:
+            self.file.write(c)
```

Comment:
I was mostly mirroring `Socket._transmit`. This method is called with a list in sys_write, the loop allows this to work with lists or strs.